### PR TITLE
ARROW-403: [Java] Create transfer pairs for internal vectors in UnionVector transfer impl

### PR DESCRIPTION
**Pull request by vkorukanti**
@StevenMPhillips, @julienledem Could you please review the patch?

### CODE REVIEW

File: java/vector/src/main/codegen/templates/UnionVector.java [hunk 1]
```diff
@@ -275,20 +269,27 @@ public FieldVector addVector(FieldVector v) {
...
-
-    UnionVector to;
-
-    public TransferImpl(String name, BufferAllocator allocator) {
-      to = new UnionVector(name, allocator, null);
+    private final TransferPair internalMapVectorTransferPair;
+    private final TransferPair typeVectorTransferPair;
+    private final UnionVector to;
+
+    public TransferImpl(String name, BufferAllocator allocator, CallBack callBack) {
```
Comment (julienledem) [inline java/vector/src/main/codegen/templates/UnionVector.java]: is the callBack here ever not null? In this PR it looks like it is always null.

**Review by julienledem — APPROVED** [review]
This overall looks good.
I just had one question.

**Comment by vkorukanti** [comment]
@julienledem: Updated the patch which removes `Callback` argument changes.

**Comment by julienledem** [comment]
LGTM. Will merge when the build passes.

**Pull request closed without merging**